platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/DataFileReader.java

```diff
@@ -168,35 +168,23 @@ public DataFileIterator createIterator() throws IOException {
         return new DataFileIterator(path, metadata, dataItemSerializer);
     }
 
-    /**
-     * Read a data item from file at dataLocation.
-     *
-     * @param dataLocation The file index combined with the offset for the starting block of the
-     *     data in the file
-     * @throws IOException If there was a problem reading from data file
-     * @throws ClosedChannelException if the data file was closed
-     */
-    public D readDataItem(final long dataLocation) throws IOException {
-        return readDataItem(dataLocation, true);
-    }
-
     /**
      * Read data item bytes from file at dataLocation and deserialize them into the Java object, if
      * requested.
      *
      * @param dataLocation The file index combined with the offset for the starting block of the
      *     data in the file
-     * @param deserialize A flag to avoid deserialization cost
      * @return Deserialized data item, or {@code null} if deserialization is not requested
      * @throws IOException If there was a problem reading from data file
      * @throws ClosedChannelException if the data file was closed
      */
-    public D readDataItem(final long dataLocation, final boolean deserialize) throws IOException {
-        final ByteBuffer data = readDataItemBytes(dataLocation);
-        return deserialize ? dataItemSerializer.deserialize(data, metadata.getSerializationVersion()) : null;
+    public D readDataItem(final long dataLocation) throws IOException {
+        long serializationVersion = metadata.getSerializationVersion();
+        final ByteBuffer data = readDataItemBytes(dataLocation, serializationVersion);
+        return dataItemSerializer.deserialize(data, serializationVersion);
     }
 
-    public ByteBuffer readDataItemBytes(final long dataLocation) throws IOException {
+    public ByteBuffer readDataItemBytes(final long dataLocation, final long serializationVersion) throws IOException {
```

Comment:
`serializationVersion` is always read from reader's own metadata, but I may be wrong. If so, there is no need to add it as a new arg, just read from metadata directly in this method.

Reply:
Fair enough.